### PR TITLE
Keep cook-executor binary updated

### PR DESCRIPTION
**Pull request by DaoWen**
Ensure cook-executor binary is rebuilt when sources change, and always
copy the newest version when building the scheduler docker image.

### CODE REVIEW

File: executor/bin/check-version.sh [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        ;;
+    '')
+        verbose=true
+        ;;
+    *)
+        echo "USAGE: $0 [-q|--quiet]"
+        exit -1
+esac
+
+source_version=`python3 -c 'from cook._version import __version__; print(__version__)'`
```
Comment (shamsimam) [inline executor/bin/check-version.sh]: Can we ensure that this command works when the script is invoked from any directory?

File: executor/bin/check-version.sh [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    fi
+    exit 1
+fi
+
+binary_app=./dist/cook-executor
+
+if [ -e $binary_app ]; then
+    installed_version="$($binary_app --version)"
+else
+    installed_version=none
```
Comment (shamsimam) [inline executor/bin/check-version.sh]: Should we eagerly return 1 here?

File: executor/cook/__main__.py [hunk 3]
```diff
@@ -21,6 +21,11 @@
 
 
 def main(args=None):
+    if len(sys.argv) == 2 and sys.argv[1] == "--version":
+        from _version import __version__
+        print(__version__)
+        sys.exit(0)
+
     print('Starting cook executor...')
```
Comment (shamsimam) [inline executor/cook/__main__.py]: Can we include the version here?

File: scheduler/bin/build-docker-image.sh [hunk 4]
```diff
@@ -3,5 +3,29 @@
...
+if $DO_EXECUTOR_REBUILD; then
+    echo "Triggering build of cook-executor before proceeding."
+    ${EXECUTOR_DIR}/bin/build-cook-executor.sh
+fi
+
+echo "Copying cook-executor from ${COOK_EXECUTOR_FILE} to ${SCHEDULER_EXECUTOR_DIR}"
+mkdir -p ${SCHEDULER_EXECUTOR_DIR}
+cp -f ${COOK_EXECUTOR_FILE} ${SCHEDULER_EXECUTOR_DIR}
+
+
```
Comment (shamsimam) [inline scheduler/bin/build-docker-image.sh]: Nitpick: delete this blank line

File: executor/bin/check-version.sh (hunk 2 above)
Comment (DaoWen) [reply executor/bin/check-version.sh]: I wanted to let this fall through to one of the `echo`s below.

File: executor/cook/__main__.py (hunk 3 above)
Comment (DaoWen) [reply executor/cook/__main__.py]: How do you want it formatted?

File: executor/bin/check-version.sh (hunk 1 above)
Comment (DaoWen) [reply executor/bin/check-version.sh]: Done

File: executor/cook/__main__.py (hunk 3 above)
Comment (shamsimam) [reply executor/cook/__main__.py]: I like: `print('Starting cook executor {}...'.format(version))`

File: executor/bin/check-version.sh [hunk 5]
```diff
@@ -23,8 +27,24 @@ case "$1" in
         exit -1
 esac
 
+#
+# Switch to executor directory
+#
+
+executor_bin_dir="$(dirname ${BASH_SOURCE[0]})"
+
+cd "$executor_bin_dir/.."
```
Comment (shamsimam) [inline executor/bin/check-version.sh]: Does this need to be a `pushd` followed by a `popd` later?

**Pull request merged**